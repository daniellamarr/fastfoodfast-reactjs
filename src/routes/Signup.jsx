import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { userSignup } from '../actions/authActions';
import Loader from '../components/Loader.jsx';

/**
 * Signup
 */
export class Signup extends Component {
  /**
   * Constructor
   */
  constructor() {
    super();

    this.state = {
      name: '',
      email: '',
      phone: '',
      address: '',
      password: '',
      cpassword: '',
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  /**
   * onChange
   * @param {object} e
   * @returns {null} null
   */
  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  /**
   * onSubmit
   * @param {object} e
   * @returns {null} null
   */
  onSubmit(e) {
    e.preventDefault();

    this.props.userSignup({
      name: this.state.name,
      email: this.state.email,
      phone: this.state.phone,
      address: this.state.address,
      password: this.state.password,
      cpassword: this.state.cpassword,
    });
  }

  /**
   * @returns {JSX} jsx
   */
  render() {
    return (
      <div id="loginPage">
        { this.props.auth.loadingStatus ? <Loader /> : null }
        <section id="login">
          <center>
            <div className="login">
              <h4>Quench your hunger, Sign up today</h4>
              <form id="signupForm" method="post" onSubmit={this.onSubmit}>
                <input
                  type="text"
                  name="name"
                  id="name"
                  placeholder="Firstname"
                  onChange={this.onChange}
                  required
                />
                <input
                  type="email"
                  name="email"
                  id="email"
                  placeholder="Email"
                  onChange={this.onChange}
                  required
                />
                <input
                  type="tel"
                  name="phone"
                  id="phone"
                  placeholder="Phone Number"
                  onChange={this.onChange}
                  required
                />
                <input
                  type="tel"
                  name="address"
                  id="address"
                  placeholder="Location"
                  onChange={this.onChange}
                  required
                />
                <input
                  type="password"
                  name="password"
                  id="password"
                  placeholder="Password"
                  onChange={this.onChange}
                  required
                />
                <input
                  type="password"
                  name="cpassword"
                  id="cpassword"
                  placeholder="Password"
                  onChange={this.onChange}
                  required
                />
                <button type="submit">Sign up </button>
              </form>
            </div>
          </center>
        </section>
        <div className="error hide"></div>
      </div>
    );
  }
}

Signup.propTypes = {
  auth: PropTypes.shape({
    loadingStatus: PropTypes.bool.isRequired,
  }),
  userSignup: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps, { userSignup })(Signup);
