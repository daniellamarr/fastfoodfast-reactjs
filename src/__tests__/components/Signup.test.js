/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { mount } from 'enzyme';
import { Signup } from '../../routes/Signup.jsx';

let wrapper;

const props = {
  auth: {
    loadingStatus: false,
  },
  userSignup: () => {},
};

beforeEach(() => {
  wrapper = mount(<Signup {...props} />);
});

describe('Loader Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should find onChange()', () => {
    wrapper.find('#email').simulate('change');
    expect(wrapper.state().email).toEqual('');
  });

  it('should find onSubmit()', () => {
    wrapper.find('#signupForm').simulate('submit', {
      preventDefault: () => {},
    });
    const props2 = { ...props, auth: { loadingStatus: true } };
    wrapper = mount(<Signup {...props2} />);
    expect(wrapper.state().email).toEqual('');
    expect(wrapper.props().auth.loadingStatus).toEqual(true);
  });
});
