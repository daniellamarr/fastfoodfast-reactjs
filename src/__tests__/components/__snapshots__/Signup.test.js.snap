// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should match snapshot 1`] = `
<Signup
  auth={
    Object {
      "loadingStatus": false,
    }
  }
  userSignup={[Function]}
>
  <div
    id="loginPage"
  >
    <section
      id="login"
    >
      <center>
        <div
          className="login"
        >
          <h4>
            Quench your hunger, Sign up today
          </h4>
          <form
            id="signupForm"
            method="post"
            onSubmit={[Function]}
          >
            <input
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="Firstname"
              required={true}
              type="text"
            />
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
            />
            <input
              id="phone"
              name="phone"
              onChange={[Function]}
              placeholder="Phone Number"
              required={true}
              type="tel"
            />
            <input
              id="address"
              name="address"
              onChange={[Function]}
              placeholder="Location"
              required={true}
              type="tel"
            />
            <input
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
            />
            <input
              id="cpassword"
              name="cpassword"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
            />
            <button
              type="submit"
            >
              Sign up 
            </button>
          </form>
        </div>
      </center>
    </section>
    <div
      className="error hide"
    />
  </div>
</Signup>
`;
